access-control-app/src/cmr/access_control/services/acl_validation.clj

```diff
@@ -93,6 +94,18 @@
                (when-not (seq (mdb1/find-concepts context {:provider-id provider-id :entry-title entry-title} :collection))
                  {key-path [(format "collection with entry-title [%s] does not exist in provider [%s]" entry-title provider-id)]})))))
 
+(defn- make-collection-concept-ids-validations
+  "Returns a validation for the concept_ids part of a collection identifier, closed over the context and ACL to be validated."
+  [context acl]
+  (let [provider-id (-> acl :catalog-item-identity :provider-id)]
+    [(v/every (fn [key-path concept-id]
+                (let [regex (re-pattern (str "C\\d+-[A-Za-z0-9_]+"))]
+                  (when-not (re-matches regex concept-id)
+                    {key-path [(format "[%s] is not a valid collection concept id." concept-id)]}))))
```

Comment:
The two messages use concept id inconsistently. I think we've been using "concept-id".